modules/images/webp-uploads/load.php

```diff
@@ -35,5 +35,186 @@ function webp_uploads_filter_image_editor_output_format( $output_format, $filena
+	$file = wp_get_original_image_path( $attachment_id, true );
+
+	// File does not exist.
+	if ( ! file_exists( $file ) ) {
+		return $metadata;
+	}
+
+	// Prevent to convert JPEG to WebP if we are creating JPEG versions of the image.
+	remove_filter( 'image_editor_output_format', 'webp_uploads_filter_image_editor_output_format' );
+
+	$sizes = array();
+	foreach ( webp_uploads_get_image_sizes() as $size => $properties ) {
+		$image_sizes = array();
+		if ( array_key_exists( 'sizes', $metadata ) && is_array( $metadata['sizes'] ) ) {
+			$image_sizes = $metadata['sizes'];
+		}
+
+		$current_size = array();
+		if ( array_key_exists( $size, $image_sizes ) && is_array( $image_sizes[ $size ] ) ) {
+			$current_size = $image_sizes[ $size ];
+		}
+
+		$sources = array();
+		if ( array_key_exists( 'sources', $current_size ) && is_array( $current_size['sources'] ) ) {
+			$sources = $current_size['sources'];
+		}
+
+		// Try to find the mime type of the image size.
+		if ( array_key_exists( 'mime-type', $current_size ) ) {
+			$current_mime = $current_size['mime-type'];
+		} elseif ( array_key_exists( 'file', $current_size ) ) {
+			$current_mime = wp_check_filetype( $current_size['file'] )['type'];
+		} else {
+			$current_mime = '';
+		}
+
+		// The mime for this file couldn't be determined.
+		if ( empty( $current_mime ) ) {
+			continue;
+		}
+
+		// Make sure the current mime is consider a valid mime type.
+		if ( ! array_key_exists( $current_mime, $valid_mime_types ) ) {
+			continue;
+		}
+
+		$sources[ $current_mime ] = array(
+			'file' => array_key_exists( 'file', $current_size ) ? $current_size['file'] : '',
+			// TOOD: Add filesize from the original version of this image.
+		);
+
+		$formats = array_diff_assoc( $valid_mime_types, array( $current_mime => $valid_mime_types[ $current_mime ] ) );
+
+		foreach ( $formats as $mime => $extension ) {
+			// Editor needs to be recreated every time as there is not flush() or clear() function that can be used after we created an image.
+			$editor = wp_get_image_editor( $file );
+
+			if ( is_wp_error( $editor ) ) {
+				continue;
+			}
+
+			$editor->resize( (int) $properties['width'], (int) $properties['height'], $properties['crop'] );
+			$filename = $editor->generate_filename( null, null, $extension );
+			$image    = $editor->save( $filename, $mime );
+
+			if ( is_wp_error( $image ) ) {
+				continue;
+			}
+
+			// TODO: Store the file size of the created image.
+			// $image['filesize'] = filesize( $image['path'] );
+			// Remove duplicated properties from the size image.
+			unset( $image['path'], $image['height'], $image['width'], $image['mime-type'] );
+
+			$sources[ $mime ] = $image;
+		}
+
+		$current_size['sources'] = $sources;
+		$sizes[ $size ]          = $current_size;
+	}
+
+	$metadata['sizes'] = $sizes;
+
+	return $metadata;
+}
+
+/**
+ * List all the available image sizes as well with the properties for each, the properties included
```

Comment:
Nit-pick, but any function doc-block descriptions in WordPress core should start with a 3rd-person verb:
```suggestion
 * Lists all the available image sizes as well with the properties for each, the properties included
```